app/js/lib/foxbox/services.js

```diff
@@ -0,0 +1,273 @@
+import EventDispatcher from './event-dispatcher';
+import BaseService from './services/base';
+import IpCameraService from './services/ip-camera';
+import LightService from './services/light';
+import DoorLockService from './services/door-lock';
+import MotionSensorService from './services/motion-sensor';
+
+const p = Object.freeze({
+  api: Symbol('api'),
+  settings: Symbol('settings'),
+  db: Symbol('db'),
+  cache: Symbol('services'),
+  isPollingEnabled: Symbol('isPollingEnabled'),
+  nextPollTimeout: Symbol('nextPollTimeout'),
+
+  // Private methods.
+  schedulePoll: Symbol('schedulePoll'),
+  getServiceInstance: Symbol('getServiceInstance'),
+  hasDoorLockChannel: Symbol('hasDoorLockChannel'),
+  updateServiceList: Symbol('updateServiceList'),
+  getCache: Symbol('getCache'),
+});
+
+/**
+ * Compare 2 objects. Returns true if all properties of object A have the same
+ * value in object B. Extraneous properties in object B are ignored.
+ * Properties order is not important.
+ *
+ * @param {Object} objectA
+ * @param {Object} objectB
+ * @return {boolean}
+ */
+const isSimilar = (objectA, objectB) => {
+  const keysA = Object.keys(objectA);
+  const keysB = Object.keys(objectB);
+
+  if (keysA.length !== keysB.length) {
+    return false;
+  }
+
+  return !keysA.some((keyA) => {
+    const valueA = objectA[keyA];
+    const valueB = objectB[keyA];
+
+    const typeA = typeof valueA;
+
+    if (typeA !== typeof valueB) {
+      return true;
+    }
+
+    if (typeA !== 'object' || valueA === null || valueB === null) {
+      return valueA !== valueB;
+    }
+
+    return !isSimilar(valueA, valueB);
+  });
+};
+
+export default class Services extends EventDispatcher {
+  constructor(db, api, settings) {
+    super(['services-changed', 'service-changed']);
+
+    this[p.db] = db;
+    this[p.api] = api;
+    this[p.settings] = settings;
+
+    this[p.cache] = null;
+    this[p.isPollingEnabled] = false;
+    this[p.nextPollTimeout] = null;
+
+    Object.seal(this);
+  }
+
+  /**
+   * Retrieves the list of the services available. Relies on IndexedDB and
+   * in-memory cache.
+   *
+   * @return {Promise<Array<BaseService>>} A promise that resolves with an
+   * array of objects.
+   */
+  getAll() {
+    return this[p.getCache]().then((cache) => Array.from(cache.values()));
```

Comment:
To properly tackle #99, I'd like to have some kind of sorting here. Maybe sort by type (lights, camera, sensors...) then by name... anything that'd make listing order consistent and predictable.
What do you think?